### PR TITLE
HT-2535: Ensure setting mfa=1 unsets IP restriction

### PR DESCRIPTION
**Pull request by aelkiss**
This works but is a little bit messy. @billdueber do you by any chance have a more idiomatic way of doing this?

### CODE REVIEW

**Comment by billdueber** [comment]
OK, so I'm confused about the `!user_params[:iprestrict]`. The PR title suggests you want to set `user_params[:iprestrict] = nil` in _all_ cases if :mfa is set. The code is setting it to nil _only_ if it's not already set. Which is it?

**Comment by aelkiss** [comment]
I think for consistency we do probably want it set to nil in all cases. I'll clean this up and submit for re-review.

File: app/controllers/ht_users_controller.rb [hunk 1]
```diff
@@ -19,7 +19,14 @@ def index
   def update
     # Any extension of term counts as a renewal for our purposes.
     renewing = user_params[:expires].present? && user_params[:expires] > @user.expires.to_date.to_s
-    if @user.update(user_params)
+
+    params = user_params
+
+    if user_params[:mfa] and !user_params[:iprestrict]
+      params = user_params.merge(iprestrict: nil)
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply app/controllers/ht_users_controller.rb:26]: Worth a try. I was having some issues where I couldn't get `@user_params` to consistently update in the way I wanted..

**Comment by moseshll** [comment]
I think the commit I just pushed fixes Bill's latest comment.

**Comment by aelkiss** [comment]
Verified on useradmin-staging; merging

**Pull request merged**